packages/checkbox/src/vaadin-checkbox.js

```diff
@@ -84,30 +84,41 @@ class Checkbox extends LabelMixin(
           align-items: baseline;
         }
 
-        .vaadin-checkbox-wrapper {
-          position: relative;
-          height: 100%;
+        [part='checkbox'],
+        ::slotted(input),
+        ::slotted(label) {
+          grid-row: 1;
+        }
+
+        [part='checkbox'],
+        ::slotted(input) {
+          grid-column: 1;
+        }
+
+        [part='checkbox'] {
+          width: var(--vaadin-checkbox-size, 1em);
+          height: var(--vaadin-checkbox-size, 1em);
+        }
+
+        [part='checkbox']::before {
+          display: block;
+          content: '\\202F';
```

Comment:
Does this make problems with screenreaders? Like we have seen with Firefox and Buttons in the past?

Reply:
It’s a "narrow non-breaking space" character. Previously it was an "em space" character, which I haven't heard would be causing any issues. I ended up copying this one from @DiegoCardoso’s branch. At one point during my prototyping it seemed like it would be better in the way that it doesn't actually increase the width of the element. Not sure if it ends up mattering in the final implementation though. But I would assume they work the same for screen readers.

I don't remember what the issue was with Firefox and Button.

Reply:
See this dicussion starting from the linked comment: https://github.com/vaadin/web-components/issues/2924#issuecomment-1318685624

Reply:
Right, now I remember reading that, thanks! 

I suppose this might look problematic? Though, it looks the same in the current release version as well.

<img width="724" alt="Screenshot 2023-01-13 at 12 59 22" src="https://user-images.githubusercontent.com/66382/212304549-8382341a-51f3-43cf-8536-2812744d9c10.png">

Reply:
The `\uea0f` is the checkmark character.

Reply:
At least for buttons it caused a problem. We don't use checkboxes heavily, so it could be possible our accessibility test didn't find it. :sweat_smile: Don't have NVDA with me to check atm, just something to consider in the future, if heavily relying on content within before/after results in weird behavior

Reply:
My assumption is that we could add `aria-hidden="true"` on `part="checkbox", so that screen readers will skip that element completely, and the pseudo elements inside it.